code/parachain/frame/composable-maths/src/dex/mod.rs

```diff
@@ -1,4 +1,33 @@
+use sp_runtime::{traits::CheckedAdd, PerThing};
+
 pub mod constant_product;
 pub mod price;
 #[cfg(test)]
 pub mod tests;
+
+pub trait WeightMath: PerThing {
+	/// Sums all weights in an array. If weights total greater than `Perthing::one()`, returns
+	/// `None`.
+	fn sum_weights(weights: &[Self]) -> Option<Self>;
+
+	/// Returns `true` if all weights in array are non-zero.
+	fn non_zero_weights(weights: &[Self]) -> bool;
+}
+
+impl<T: PerThing + CheckedAdd> WeightMath for T {
+	fn sum_weights(weights: &[Self]) -> Option<Self> {
+		weights
+			.iter()
+			.try_fold(Self::zero(), |total_weight, weight_n| total_weight.checked_add(weight_n))
+	}
+
+	fn non_zero_weights(weights: &[Self]) -> bool {
+		for weight in weights {
```

Comment:
We can probably combine the two functions into one with an error return if a weight is zero as we don't expect any weights to be zero in any case

Reply:
While I agree it would be a slight performance optimization, I wanted to keep these functions with single responsibility. While the `non_zero_weights` function is likely only to be used in one context, the `sum_weights` function has some more generic utility.